src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -299,6 +299,12 @@ static int8_t CDC_Itf_Receive(uint8_t* Buf, uint32_t *Len)
 {
     rxAvailable = *Len;
     rxBuffPtr = Buf;
+    if (!rxAvailable || !rxBuffPtr) {
```

Comment:
Is there a reason to not add this in `CDC_Receive_DATA()` where `USBD_CDC_ReceivePacket()` is normally called for non-zero-length packets?

Also, to my knowledge the only safe assumption about the zero-length packet is that `len` is 0 - not sure if different versions of the STM libraries might return a non-null `Buf`.

Reply:
> Is there a reason to not add this in `CDC_Receive_DATA()` where `USBD_CDC_ReceivePacket()` is normally called for non-zero-length packets?

`CDC_Receive_DATA()` is only called by the application layer if `CDC_Receive_BytesAvailable()` returns that there are bytes to process (`rxAvailable`). Since `rxAvailable` is zero we never actually call `CDC_Receive_DATA()` in this situation. I actually tried to do it the way you're suggesting originally and found these problems.

> Also, to my knowledge the only safe assumption about the zero-length packet is that `len` is 0 - not sure if different versions of the STM libraries might return a non-null `Buf`.

The code is written with an OR. It's probably a good check in general to make sure the passed buffer is non-NULL.

Reply:
Ok, then we have to do this here. But not sure if calling `USBD_CDC_ReceivePacket()` here if `len` isn't 0 (i.e. `!rxBuffPtr`) is improving things - this should probably never happen (unless it is intentional and used to signal some special condition), but if we wanted to detect and prevent NULL access here then we should rather do something like:
```
if (rxAvailable && !rxBuffPtr) {
    rxAvailable = 0;
}
```
But my hunch is to only fix the bug that we know of and assume the rest of the library code is by design.

Reply:
Ok, changed it to only check for the `Len` being 0.